### PR TITLE
[PyTorch] Convert exported programs without run_decompositions

### PR DESCRIPTION
**Pull request by mvafin**
### Details:

- Convert `torch.export` models without `run_decompositions`, avoiding extra AOTAutograd tracing and shape propagation.
- Reuse TorchScript converters through ATen-name fallback and centralize FX multi-output handling, reducing the FX table from 343 entries to 125.
- Share TorchScript/FX alias and mutation handling for views, containers, metadata changes, and nested graphs. Replace TorchScript alias exclusions with schema-based checks and enable mutation/`out=` regression tests.
- Use torch 2.14 / torchvision 0.29 for PyTorch tests. Keep shared constraints and LLM accuracy on torch 2.13 / torchvision 0.28 for WhoWhatBench compatibility, and GPTQ on torch 2.3.1.

Validation on CPU: 16,505 TorchScript, 12,954 export, 5,170 torch.compile, and 729 frontend/API tests passed, plus RetinaNet, GPT-2, and GPTQ checks. Build, style, dependency resolution, and dependency review passed. The ARM FP32 configuration was exercised on x86; native ARM confirmation remains pending.

### Tickets:

- CVS-194228
- #37626

### AI Assistance:

- AI assistance used: yes.
- Codex implemented and reviewed the changes and ran builds, tests, and style checks. Human review and validation are pending.

### CODE REVIEW

File: src/frontends/pytorch/src/op/rand.cpp [hunk 1]
```diff
@@ -285,7 +285,13 @@ OutputVector translate_normal(const NodeContext& context) {
...
     auto std = context.get_input(1);
     auto dtype = element::f32;
-    if (context.get_input_size() == 3 || context.get_input_size() == 4) {
+    if (context.get_op_type().find("aten.normal.float_float") == 0) {
+        if (context.has_attribute("dtype")) {
+            dtype = context.get_attribute<element::Type>("dtype");
+        }
+        return make_random_normal(context, context.get_input(2), dtype, std, mean);
+    }
+    if (context.get_input_size() >= 2 && context.get_input_size() <= 4) {
```
[the remark this answers is not in the document]
Comment (mvafin) [reply src/frontends/pytorch/src/op/rand.cpp:294]: Fixed in baa7e8480f. The shared converter now takes the random-sample shape from `std` when `mean` is a Python scalar, and from `mean` otherwise.

Added four CPU precommit regressions covering one- and two-dimensional tensors, with and without `out=`, in both export and TorchScript paths. They check shape, finite values, and variation between elements, since a scalar sample broadcast over `std` can still produce the expected output shape. All four export cases failed before the fix and pass afterward.

Validation: 11 affected export tests and 22 TorchScript tests passed on CPU/FP32 with PyTorch 2.12.1. Release build, clang-tidy, frontend naming checks, clang-format, and copyright checks passed.

File: tests/constraints.txt [hunk 2]
```diff
@@ -25,5 +25,5 @@ pytest-timeout==2.4.0
 kornia==0.8.3
 
 --extra-index-url https://download.pytorch.org/whl/cpu
-torch~=2.12.0
-torchvision~=0.27.0
+torch~=2.14.0
+torchvision~=0.29.0
```
[the remark this answers is not in the document]
Comment (mvafin) [reply tests/constraints.txt]: Fixed in 23114638ad. The LLM accuracy suite now pins `torch==2.12.0` and `torchvision==0.27.0`; the shared torch 2.14 / torchvision 0.29 requirements and GPTQ pins remain unchanged.

Extracted the shared non-PyTorch constraints into `tests/constraints_common.txt` so the LLM suite retains those limits. Updated both test-package installation and the LLM job sparse checkout to include the new file.

Validation: clean Python 3.12 pip dry-run resolved all 201 dependencies, including WhoWhatBench. Also verified that expanding the shared constraints produces exactly the prior constraints, and checked CMake test-package installation and workflow YAML/sparse checkout paths.

File: tests/llm/requirements.txt [hunk 3]
```diff
@@ -1,5 +1,8 @@
 -c ../constraints.txt
 --extra-index-url https://download.pytorch.org/whl/cpu
+# torch 2.13.0 fixes GHSA-rrmf-rvhw-rf47; WhoWhatBench caps torchvision at 0.28.0.
+torch==2.13.0
+torchvision==0.28.0
```
Comment (mvafin) [inline tests/llm/requirements.txt]:
```suggestion
```

File: src/bindings/python/src/openvino/frontend/pytorch/fx_decoder.py [hunk 4]
```diff
@@ -279,23 +336,6 @@ def from_exported_program(
             to the ``target_op`` that the C++ frontend expects for
             ``ConversionExtension`` lookup.
         """
-        from packaging import version
```
Comment (evkotov) [inline src/bindings/python/src/openvino/frontend/pytorch/fx_decoder.py:282]: The removal of `run_decompositions()` from `TorchFXPythonDecoder.from_exported_program` (and of `get_export_decomposition_list()`/`ops_to_not_decompose()`/export_decompositions.py) drops support for a number of ATen ops that never had a direct TorchScript or FX translator and previously only worked because decomposition expanded them into primitives with existing translators — e.g. diag_embed, huber_loss, renorm, trace, unfold, xlogy, nan_to_num, block_diag, isin, mse_loss/l1_loss/multi_margin_loss, softshrink, count_nonzero, sinc, lerp/logaddexp2, and others. The new ATen-name fallback (normalize_op_type) only helps when a matching translator already exists under the normalized name, which is not the case for these ops. Models converted via torch.export that use any of them will now fail with "no translator found" instead of converting successfully. Could you confirm this is intentional and add coverage tests for at least a subset of these ops, or clarify the migration path for them?

File: src/frontends/pytorch/src/op/norm.cpp [hunk 5]
```diff
@@ -54,9 +54,6 @@ Output<Node> norm_vector(const NodeContext& context,
         auto abs = context.mark_node(std::make_shared<v0::Abs>(input_tensor));
         res = context.mark_node(std::make_shared<v1::ReduceMin>(abs, dim, keep_dim));
     } else if (p == 0) {
-        auto input_rank = input_tensor.get_partial_shape().rank();
```
Comment (evkotov) [inline src/frontends/pytorch/src/op/norm.cpp:57]: The `PYTORCH_OP_CONVERSION_CHECK(input_rank.is_dynamic() || input_rank.get_length() == 1, "ord=0 supported only for vector norm")` guard was removed from norm_vector's p==0 branch with no equivalent check added elsewhere, while translate_linalg_vector_norm/translate_norm/translate_linalg_norm still call norm_vector with ord=0 for tensors of any rank. If the rank-1 restriction was there for a real correctness reason, e.g. torch.linalg.vector_norm(x, ord=0, dim=1) on a rank>1 tensor will now silently produce a result without a check, and no test exercises ord=0 on a multi-dimensional tensor. Could you clarify whether this removal is an intentional fix (the NotEqual+ReduceSum implementation is correct per-axis for any rank) and add a regression test for ord=0 with rank>1?

File: src/bindings/python/src/openvino/frontend/pytorch/fx_decoder.py (hunk 4 above)
Comment (mvafin) [reply src/bindings/python/src/openvino/frontend/pytorch/fx_decoder.py:282]: It is intentional and a drawback we have to take if we want to remove decompositions. Since tests are passing it means we didn't have those ops previously in models and didn't add tests for them. If such models will appear we will ahve to add those ops support in both TS and FX path.

File: src/frontends/pytorch/src/op/norm.cpp (hunk 5 above)
Comment (mvafin) [reply src/frontends/pytorch/src/op/norm.cpp:57]: The rank-check removal is intentional: `ord=0` counts nonzero elements along the selected reduction axes, so the existing NotEqual + ConvertLike + ReduceSum implementation supports multidimensional inputs.

Added regression coverage in cf4a41d157 using zeros and positive/negative values in 2D and 3D tensors. It covers dim=1, a negative axis, multiple axes, dim=None, both keepdim settings, and static/dynamic shapes through TorchScript and torch.export.

Validation after merging master (a12309b5ba): Release build of the PyTorch frontend, Python bindings, and CPU plugin passed; 20 new norm cases plus 4 scalar-mean normal regressions passed in each conversion mode, and all 41 decoder tests passed (89 tests total, CPU/FP32, PyTorch 2.13.0).

**Review by evkotov — APPROVED** [review]
